### ISSUE TITLE
LINK : fatal error LNK1104: cannot open file

### ISSUE THREAD

**Issue by andre2007** [intellij-dlanguage/intellij-dlanguage#452]
While changing a lot of source code and starting the dub application within IntelliJ there are often errors like:
LINK : fatal error LNK1104: cannot open file '.dub\build\application-debug-windows-x86_64-dmd_2081-51401676483BA91CBCCF20A2B9868C40\ansys-twin-shell.ilk'
Error: linker exited with status 1104
C:\D\dmd2\windows\bin\dmd.exe failed with exit code 1.

The error is solved by just starting again the application. I would like to understand what exactly happens. Does IntelliJ D plugin compiles the source code in the background to check for syntax errors?
If yes, where can I find this code? I searched but didn't found the location.

**Comment by andre2007** [comment]
I found the java file io.github.intellij.dlanguage.highlighting.annotation.external.

On every code change, dub will be triggered to compile a new executable just for checking the syntax. This causes several issues:
- Performance
- SSD lifetime?
- File access error like the one mentioned

DMD supports the flag "-o-" which can be used to trigger syntax check only without generating a binary. I created a pull request for dub to add the build type "syntax" which will lead exactly to the flag "-o-".
https://github.com/dlang/dub/pull/1514

In the meantime the flag can be passed to DMD using the environment  variable DFLAGS. I will check whether I can create a pull request.
